samples/canopen/canopen_rpdo/TestCanNode.hpp

```diff
@@ -0,0 +1,170 @@
+     *
+     * @return Pointer to the start of the object dictionary
+     */
+    CO_OBJ_T* getObjectDictionary();
+
+    /**
+     * Get the number of elements in the object dictionary.
+     *
+     * @return The number of elements in the object dictionary
+     */
+    uint8_t getNumElements();
+
+private:
+    /**
+     * The node ID used to identify the device on the CAN network.
+     */
+    static constexpr uint8_t NODE_ID = 0x02;
+
+    /**
+     * This sample data will be exposed over CAN through the object
+     * dictionary. The address of the variable will be included in the
+     * object dictionary and can be updated via SDO via a CANopen client.
+     * This device will then broadcast the value via a triggered PDO.
+     */
+    uint8_t sampleDataA;
+    uint16_t sampleDataB;
+
+    /**
+     * Have to know the size of the object dictionary for initialization
+     * process.
+     */
+    static constexpr uint8_t OBJECT_DICTIONARY_SIZE = 15;
+
+    /**
+     * The object dictionary itself. Will be populated by this object during
+     * construction.
+     *
+     * The plus one is for the special "end of dictionary" marker.
+     */
+    CO_OBJ_T objectDictionary[OBJECT_DICTIONARY_SIZE + 1] = {
+        // Sync ID, defaults to 0x80
+        {CO_KEY(0x1005, 0, CO_UNSIGNED32 | CO_OBJ_D__R_), 0, (uintptr_t) 0x80},
+
+        // Information about the hardware, hard coded sample values for now
+        // 1: Vendor ID
+        // 2: Product Code
+        // 3: Revision Number
+        // 4: Serial Number
+        {
+            .Key = CO_KEY(0x1018, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = 0,
+            .Data = (uintptr_t) 0x10,
+        },
+        {
+            .Key = CO_KEY(0x1018, 2, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = 0,
+            .Data = (uintptr_t) 0x11,
+        },
+        {
+            .Key = CO_KEY(0x1018, 3, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = 0,
+            .Data = (uintptr_t) 0x12,
+        },
+        {
+            .Key = CO_KEY(0x1018, 4, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = 0,
+            .Data = (uintptr_t) 0x13,
+        },
+
+        // SDO CAN message IDS.
+        // 1: Client -> Server ID, default is 0x600 + NODE_ID
+        // 2: Server -> Client ID, default is 0x580 + NODE_ID
+        {
+            .Key = CO_KEY(0x1200, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = 0,
+            .Data = (uintptr_t) 0x600 + NODE_ID,
+        },
+        {
+            .Key = CO_KEY(0x1200, 2, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = 0,
+            .Data = (uintptr_t) 0x580 + NODE_ID,
+        },
+
+        //RPDO settings
+        // 0: RPDO number in index and total number of sub indexes.
+        // 1: The COB-ID to receive PDOs from.
+        // 2: transmission trigger
+        {
+            .Key = CO_KEY(0x1400, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
+            .Type = 0,
+            .Data = (uintptr_t) 3},
```

Comment:
If you add commas at the end of the `.Data` lines, clang-format will realize that the `}` should go on the next line instead of at the end of the current line. Just for the sake of formatting, can you go through and add the commas to this and the following lines? It's easiest to find the missing commas if you run clang-format locally